include/ranges.h

```diff
@@ -471,6 +503,9 @@ struct chunk {
 	chunk(const id<Dims>& offset, const celerity::range<Dims>& range, const celerity::range<Dims>& global_size)
 	    : offset(offset), range(range), global_size(global_size) {}
 
+	/// Returns the smallest dimensionality that `*this` can be `chunk_cast` to.
+	int get_min_dimensions() const { return std::max({offset.get_min_dimensions(), range.get_min_dimensions(), global_size.get_min_dimensions()}); }
```

Comment:
`box` and `region` have a `static constexpr int dimensions` member (as does `sycl::range` / `sycl::id`), and this is what I intended the word "dimensions" to refer to. But since we're bikeshedding, we could also name it `get_effective_dimensions` or `get_effective_dims` because _effective dimensionality_ is the terminology used in grid.cc.

Reply:
I (perhaps unreasonably strongly, but still) dislike the `s` at the end of those proposed names. My expectation when reading a `get_...s()` is that I get more than one thing. I'd be perfectly fine with `get_effective_dim`.

Reply:
Renamed to `get_effective_dims` now for consistency. I see your point @PeterTh but we use `int Dims` excessively throughout the codebase. The naming stems from SYCL originally. I'm not in principle opposition to rename this globally.

Reply:
I think `int Dims` is a bit different since it doesn't have the getter implication of being more than one thing, but I see the point about consistency.